docs/cookbooks/deploying_aws.md

```diff
@@ -0,0 +1,549 @@
+```
+
+Lastly you can deploy your ECS service by scaling up the container count using the AWS CLI:
+
+```bash
+aws ecs update-service --cluster gr-backend-ecs-cluster --service gr-backend-ecs-service --desired-count 3 --force-new-deployment
+```
+
+
+# Step 3: (Optional) CI
+
+In some cases you may update your `config.py` file or your `hub-requirements.txt` file to add/remove guards. Here it may help to set up CI to automate the process of updating your Guardrails API container and deploy to ECS.
+
+For this to work you must set the following configuration on Github Actions:
+- `AWS_ACCESS_KEY_ID` [Secret]
+- `AWS_SECRET_ACCESS_KEY` [Secret]
+- `AWS_ECR_REPOSITORY` [Variable]
+- `AWS_ECS_CLUSTER_NAME` [Variable] Can be set to `gr-backend-ecs-cluster` as default
+- `AWS_ECS_SERVICE_NAME` [Variable] Can be set to `gr-backend-ecs-service` as default
+
+Here is some provided Github Actions workflow which does this on pushes to the `main` branch:
+
+
+```yaml
+name: Deploy Backend
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy_app:
+    name: Deploy Backend
+    runs-on: ubuntu-latest
+    env:
+      AWS_REGION: us-east-1
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      AWS_ECR_REPOSITORY: ${{ vars.AWS_ECR_REPOSITORY }} 
+      AWS_ECS_CLUSTER_NAME: ${{ vars.AWS_ECS_CLUSTER_NAME }}
+      AWS_ECS_SERVICE_NAME: ${{ vars.AWS_ECS_SERVICE_NAME }}
+      WORKING_DIR: "./"
+    steps: 
+    
+      - name: Check out head
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@master
+        with:
+          platforms: linux/amd64
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@master
+        with:
+          platforms: linux/amd64
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: ${{ env.AWS_REGION }}
+          aws-secret-access-key: ${{ env.AWS_SECRET_ACCESS_KEY }}
+          aws-access-key-id: ${{ env.AWS_ACCESS_KEY_ID }}
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Build & Push ECR Image
+        uses: docker/build-push-action@v2
+        with:
+          builder: ${{ steps.buildx.outputs.name }}
+          context: ${{ env.WORKING_DIR }}
+          platforms: linux/amd64
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          push: true
+          tags: ${{ env.AWS_ECR_REPOSITORY }}:latest
+
+      - name: Deploy to ECS
+        run: |
+          aws ecs update-service --cluster ${{ env.AWS_ECS_CLUSTER_NAME }} --service ${{ env.AWS_ECS_SERVICE_NAME }} --desired-count 1 --force-new-deployment
+        env:
+          AWS_DEFAULT_REGION: ${{ env.AWS_REGION }}
+
+
+```
+
+# Terraform
+
+```hcl
+locals {
```

Comment:
maybe not for this iteration but does it make sense to wrap this in a terraform module to get these resources into their own name space and reusable?

Reply:
yup! in theory this can already be dropped into a subfolder to be used as a module as is

Reply:
I'll a snippet in the cookbook mentioning this 👍🏼